### PR TITLE
feat: add extension points in components

### PR DESCRIPTION
**Pull request by fred-r**
Update the schema and documentation to support
the extension points defined in the Open-CMSIS-Pack forum.

### CODE REVIEW

**Comment by fred-r** [comment]
fixes #27 
fixes #23

File: schema/PACK.xsd [hunk 1]
```diff
@@ -17,14 +17,16 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 
-  $Date:        21. Apr 2022
-  $Revision:    1.7.7
+  $Date:        01.06.2022
```
Comment (JonatanAntoni) [inline schema/PACK.xsd:20]: The date must be kept in the valid format
```suggestion
  $Date:        01. Jun 2022
```

File: schema/PACK.xsd [hunk 2]
```diff
@@ -17,14 +17,16 @@
...
+  $Revision:    1.7.8
 
   $Project: Schema File for Package Description File Format Specification
 
   Package file name convention <vendor>.<name>.<version>.pack
+  SchemaVersion=1.7.8
 
-  SchemaVersion=1.7.7
+  1. June 2022:
```
Comment (JonatanAntoni) [inline schema/PACK.xsd:28]: Version is missing, please keep the format:
```suggestion
  1. June 2022: v1.7.8
```

**Comment by JonatanAntoni** [comment]
Please add new version to version history in documentation:
https://github.com/Open-CMSIS-Pack/Open-CMSIS-Pack-Spec/blob/a27669c88fdb20842ede9c5094b0713c3ab4e5a3/doxygen/src/General.txt#L86

**Review by JonatanAntoni — CHANGES_REQUESTED** [review]

File: schema/PACK.xsd (hunk 1 above)
Comment (fred-r) [reply schema/PACK.xsd]: Done

File: schema/PACK.xsd (hunk 2 above)
Comment (fred-r) [reply schema/PACK.xsd]: Done

**Review by JonatanAntoni — APPROVED** [review]

**Pull request merged**